### PR TITLE
Add functionality to parse multiple values within a single Observation from `component`

### LINKED ISSUE #1304: Ensure parsing multiple values within an Observation
See Exposure Observation example where the component section is not being properly parsed.

Acceptance Criteria:
All appropriate Components and valueCodeableConcepts are included in a PHDC observation

### PR DESCRIPTION
**Pull request by m-goggins**
# PULL REQUEST

## Summary
This PR adds functionality for parsing multiple observations within an Observation resource, e.g., observations within the `component` of an Observation resource. Each observation is parsed as a "tertiary schema" from the message parsing schema and added as a separate observation entry in the PHDC.

## Related Issue
Fixes #1304 

## Additional Information
Anything else the review team should know?

## Checklist

- [ ] If this code affects the other scrum team, have they been notified? (In Slack, as reviewers, etc.)

[//]: # (PR title: Remember to name your PR descriptively!)

### CODE REVIEW

**Review by DanPaseltiner — CHANGES_REQUESTED** [review]
Thanks for thinking through all of this @m-goggins! I think there is some room to simplify and streamline things a bit (see comments below).

File: containers/message-parser/app/default_schemas/phdc_case_report_schema.json [hunk 1]
```diff
@@ -317,6 +317,63 @@
...
+          "component_value_qualitative_code": {
+            "fhir_path": "valueCodeableConcept.coding[0].code",
+            "data_type": "string",
+            "nullable": true
+          },
+          "component_text": {
+            "fhir_path": "valueCodeableConcept.text",
+            "data_type": "string",
+            "nullable": true
+          }
```
Comment (DanPaseltiner) [inline containers/message-parser/app/default_schemas/phdc_case_report_schema.json:375]: If we make all of these field names (keys) consistent with the level above when we unpack into our internal `Observation` objects we should be able to reuse the existing attributes and no longer need to add attributes specific to data from `component`.

File: containers/message-parser/app/phdc/models.py [hunk 2]
```diff
@@ -140,9 +140,21 @@ class Observation:
...
+    component_code_code_system: Optional[str] = None
+    component_code_code_display: Optional[str] = None
+    component_value_quantitative_value: Optional[str] = None
+    component_value_quant_code_system: Optional[str] = None
+    component_value_quantitative_code: Optional[float] = None
+    component_value_qualitative_value: Optional[str] = None
+    component_value_qualitative_code_system: Optional[str] = None
+    component_value_qualitative_code: Optional[str] = None
+    component_text: Optional[str] = None
+    component_bool: Optional[bool] = False
```
[the remark this answers is not in the document]
Comment (m-goggins) [reply containers/message-parser/app/phdc/models.py:157]: That's a good point! I'll make the change

**Review by DanPaseltiner — APPROVED** [review]
🚀  Looks great! Thanks for identifying the need for this and implementing a solution.

**Review by bamader — APPROVED** [review]
Great stuff here, Marcelle! Thanks for getting this all together.

**Pull request merged**